### PR TITLE
Design doc for portable console switch for multiple vendors

### PR DESCRIPTION
**Pull request by lizhijianrd**
In this design document, we provide the standard for adapting console devices to SONiC. If a vendor's console device implementation conforms to this standard, it will work on any switch with SONiC installed.

### CODE REVIEW

File: doc/console/Portable-Console-Device-High-Level-Design.md [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+  ```
+
+#### show console serial_number
+
+This command displays the serial number of the portable console device.
+
+* Usage:
+
+  ```
+  show console serial_number
```
Comment (Blueve) [inline doc/console/Portable-Console-Device-High-Level-Design.md:320]: **From Xin's comments:**

Need to consider daisy-chain, we need to have all chained devices information.

File: doc/console/Portable-Console-Device-High-Level-Design.md [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+### CONSOLE_SWITCH_TABLE
+
+The `CONSOLE_SWITCH_TABLE` holds the configuration database for the purpose of console switch features. This table is filled by the management framework. Three new fields will be added to this table.
+
+```
+; Console switch feature table
+key = CONSOLE_SWITCH:console_mgmt
+
+; field = value
+autodetect  = "enable"/"disable" ; "enabled" means factory function will auto detect which vendor's device is plugged in
```
Comment (Blueve) [inline doc/console/Portable-Console-Device-High-Level-Design.md:52]: From Xin's comment:

We need add restrictions for these value. Such as, if autodetect is enabled, then vendor_name/model_name must be empty

File: doc/console/Portable-Console-Device-High-Level-Design.md (hunk 1 above)
Comment (lizhijianrd) [reply doc/console/Portable-Console-Device-High-Level-Design.md:374]: Thanks, I support daisy-chain in API design in this commit: [98aa6bc](https://github.com/sonic-net/SONiC/pull/1012/commits/98aa6bc59e7554fa18560b4ec341a626146521fc).

File: doc/console/Portable-Console-Device-High-Level-Design.md (hunk 2 above)
Comment (lizhijianrd) [reply doc/console/Portable-Console-Device-High-Level-Design.md:52]: Do we need to mention it in this design doc? Or simply add restrictions when coding?

Comment (lizhijianrd) [reply doc/console/Portable-Console-Device-High-Level-Design.md:52]: Resolved in latest version.

**Comment by zhangyanzhao** [comment]
@lizhijianrd several things need be updated:

1. Please define YANG table for the new config_DB table
2. Please add the new CLIs to  https://github.com/sonic-net/sonic-utilities/blob/master/doc/Command-Reference.md
3. Please add the code PRs into this HLD by referring to [PR#806](https://github.com/Azure/SONiC/pull/806)

File: doc/console/Portable-Console-Device-High-Level-Design.md [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+
+        :return: An integer, the number of console lines on portable console devices.
+        """
+        raise NotImplementedError
+
+    def get_all_lines(self):
+        """
+        Retrieves the infomation of all console lines on portable console devices.
+
+        :return: A dict, the key is console line number (integer, same as portable console device designed),
```
Comment (Blueve) [inline doc/console/Portable-Console-Device-High-Level-Design.md:199]: If you daisy-chain 2 same console devices, and this model have 8ports. How the port name of 2nd console devices aligned with physical port name?

**Review by Blueve — APPROVED** [review]

**Comment by lizhijianrd** [comment]
> @lizhijianrd several things need be updated:
> 
> 1. Please define YANG table for the new config_DB table
> 2. Please add the new CLIs to  https://github.com/sonic-net/sonic-utilities/blob/master/doc/Command-Reference.md
> 3. Please add the code PRs into this HLD by referring to [PR#806](https://github.com/Azure/SONiC/pull/806)
> 4. Please make sure to have the UT defined

Thanks @zhangyanzhao! Currently we are still in the design phase, so I'll update these points when I begin to implementation.

**Pull request merged**